app/uk/gov/hmrc/taxenrolmentassignmentfrontend/services/ThrottlingService.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.taxenrolmentassignmentfrontend.services
+
+import cats.data.EitherT
+import uk.gov.hmrc.auth.core.{Enrolment, EnrolmentIdentifier, Enrolments}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.service.TEAFResult
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.AccountTypes
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.AccountTypes.SINGLE_ACCOUNT
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.config.AppConfig
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.connectors.LegacyAuthConnector
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.models.enums.EnrolmentEnum.hmrcPTKey
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+sealed trait ThrottleResult
+case object ThrottleApplied extends ThrottleResult
+case object ThrottleDoesNotApply extends ThrottleResult
+
+class ThrottlingService @Inject()(legacyAuthConnector: LegacyAuthConnector, appConfig: AppConfig) {
+
+  def throttle(accountType: AccountTypes.Value, nino: String, currentEnrolments: Set[Enrolment])(
```

Comment:
Should we not be saving the throttle result to cache? Then as a first step see if it is in the cache and if not then do the logic?

Reply:
the check of PT_ASSIGNED_TO_CURRENT_USER should catch this scenario, I'm unsure how much value a call to mongo gives us as its got a TTL and we wont be doing any other calls to AUTH IF user is PT_ASSIGNED_TO_CURRENT_USER